tests/test_schemas.py

```diff
@@ -797,27 +852,67 @@ def test_manually_routing_generic_view(self):
 
         generator = SchemaGenerator(title='Naming Colisions', patterns=patterns)
 
-        with pytest.raises(ValueError):
-            generator.get_schema()
+        schema = generator.get_schema()
+
+        self._verify_cbv_links(schema['test']['delete'], '/test/delete/')
+        self._verify_cbv_links(schema['test']['put'], '/test/put/')
+        self._verify_cbv_links(schema['test']['get'], '/test/get/')
+        self._verify_cbv_links(schema['test']['update'], '/test/update/')
+        self._verify_cbv_links(schema['test']['retrieve'], '/test/retrieve/')
+        self._verify_cbv_links(schema['test'], '/test')
 
     def test_from_router(self):
         patterns = [
             url(r'from-router', include(naming_collisions_router.urls)),
         ]
 
         generator = SchemaGenerator(title='Naming Colisions', patterns=patterns)
+        schema = generator.get_schema()
+        desc = schema['detail_0'].description  # not important here
+
+        expected = coreapi.Document(
+            url='',
+            title='Naming Colisions',
+            content={
+                'detail': {
+                    'detail_export': {},
+                    'detail_export_0': coreapi.Link(
+                        url='/from-routercollision/detail/export/',
+                        action='get',
+                        description=desc)
+                },
+                'detail_0': coreapi.Link(
+                    url='/from-routercollision/detail/',
+                    action='get',
+                    description=desc
+                )
+            }
+        )
+
+        assert schema == expected
+
+    def test_url_under_same_key_not_replaced(self):
+        patterns = [
+            url(r'example/(?P<pk>\d+)/$', BasicNamingCollisionView.as_view()),
+            url(r'example/(?P<slug>\w+)/$', BasicNamingCollisionView.as_view()),
+        ]
+
+        generator = SchemaGenerator(title='Naming Colisions', patterns=patterns)
+        schema = generator.get_schema()
 
-        with pytest.raises(ValueError):
-            generator.get_schema()
+        assert schema['example']['read'] == {}
+        assert schema['example']['read_0'].url == '/example/{id}/'
+        assert schema['example']['read_1'].url == '/example/{slug}/'
 
+    def test_url_under_same_key_not_replaced_another(self):
 
-def test_is_list_view_recognises_retrieve_view_subclasses():
```

Comment:
Why have you removed this test case? I assume it's a mistake...

It's for a separate issue. #5165